### PR TITLE
Add stateDelta to process step in graphql

### PR DESCRIPTION
**Pull request by tjeerddie**
`deepdiff` is good to see the differences but not for getting the new and updated values and you would need to parse that data. There is a [dictdiffer](https://pypi.org/project/dictdiffer/) package that gives it in a better format which seems much easier to parse into new and updated values.

I kept this simple to get the same step state delta as frontend V1 creates but it's much faster.

The delta of one of our surf processes took frontend 30sec and backend 0,248 seconds

Closes: #375

### CODE REVIEW

File: orchestrator/utils/get_updated_fields.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+
+def get_dict_updates(obj1: dict, obj2: dict, delta: bool = False) -> dict:
```
Comment (mrijk) [inline orchestrator/utils/get_updated_fields.py:17]: Mixed use of dict and Dict in the orchestrator-core codebase

File: orchestrator/utils/get_updated_fields.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            elif obj1[key] != obj2[key]:
+                updated[key] = obj2[key]
+        else:
+            updated[key] = obj2[key]
+    return updated
+
+
+def get_updated_fields(obj1: Any, obj2: Any, delta: bool = False) -> dict | Any | None:
+    if isinstance(obj1, dict) and isinstance(obj2, dict):
+        return get_dict_updates(obj1, obj2, delta)
```
Comment (mrijk) [inline orchestrator/utils/get_updated_fields.py:34]: Looks like mutual recursion. I didn't check, but wasn't it possible to write in a single function?

File: orchestrator/utils/get_updated_fields.py (hunk 1 above)
Comment (tjeerddie) [reply orchestrator/utils/get_updated_fields.py:17]: yea, 3.9 -_-

File: orchestrator/utils/get_updated_fields.py (hunk 2 above)
Comment (tjeerddie) [reply orchestrator/utils/get_updated_fields.py:34]: I think it is easier to go through an entire dict with recursion for the updated props.

File: orchestrator/utils/enrich_process.py [hunk 3]
```diff
@@ -11,14 +11,17 @@
...
+from more_itertools import first
+
+from orchestrator.db import ProcessStepTable, ProcessTable, SubscriptionTable
+from orchestrator.utils.get_updated_properties import get_dict_updates
+from orchestrator.workflow import ProcessStat, Step, StepStatus
 from pydantic_forms.core import generate_form
 
 
-def format_subscription(subscription: SubscriptionTable) -> dict:
+def format_subscription(subscription: SubscriptionTable) -> Dict:
```
Comment (dmarjenburgh) [inline orchestrator/utils/enrich_process.py:24]: We require python >= 3.9. `dict` is equally powerful as `Dict` and preferred

File: orchestrator/utils/get_updated_properties.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Union
+
+
+def get_dict_updates(obj1: Dict, obj2: Dict) -> Dict:
```
Comment (dmarjenburgh) [inline orchestrator/utils/get_updated_properties.py:17]: Use plain `dict` instead of `Dict`

File: orchestrator/utils/get_updated_properties.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                if nested_updates:
+                    updated[key] = nested_updates
+            elif obj1[key] != obj2[key]:
+                updated[key] = obj2[key]
+        else:
+            updated[key] = obj2[key]
+    return updated
+
+
+def get_updated_properties(obj1: Any, obj2: Any) -> Union[Dict, Any, None]:
```
Comment (dmarjenburgh) [inline orchestrator/utils/get_updated_properties.py:32]: `Union` with `Any` is just `Any`

File: orchestrator/utils/enrich_process.py (hunk 3 above)
Comment (tjeerddie) [reply orchestrator/utils/enrich_process.py:24]: ye, thought 3.9 didn't like it....

**Pull request merged**